### PR TITLE
CHE-1219: add heartbeat messages to keep websocket connection alive

### PR DESCRIPTION
**Pull request by ashumilova**
Signed-off-by: Ann Shumilova ashumilova@codenvy.com

### CODE REVIEW

File: dashboard/src/components/api/che-websocket.factory.js [hunk 1]
```diff
@@ -22,9 +22,10 @@ export class CheWebsocket {
    * Default constructor that is using resource
    * @ngInject for Dependency injection
    */
-  constructor ($websocket, $location, proxySettings, userDashboardConfig) {
+  constructor ($websocket, $location, $interval, proxySettings, userDashboardConfig) {
```
Comment (garagatyi) [inline dashboard/src/components/api/che-websocket.factory.js:25]: I think it is better to explain what `interval` means. You can rename it to pingInterval

File: dashboard/src/components/api/che-websocket.factory.js [hunk 2]
```diff
@@ -145,15 +156,45 @@ class MessageBuilder {
 
 class MessageBus { // jshint ignore:line
 
-  constructor(datastream) {
+  constructor(datastream, $interval) {
```
Comment (garagatyi) [inline dashboard/src/components/api/che-websocket.factory.js:159]: I think it is better to explain what interval means. You can rename it to pingInterval

File: dashboard/src/components/api/che-websocket.factory.js (hunk 1 above)
Comment (ashumilova) [inline dashboard/src/components/api/che-websocket.factory.js:25]: it's AngularJs common wrapper for JS Window.setInterval ()

Comment (benoitf) [inline dashboard/src/components/api/che-websocket.factory.js:25]: yes these are angularjs variables

File: dashboard/src/components/api/che-websocket.factory.js (hunk 2 above)
Comment (benoitf) [inline dashboard/src/components/api/che-websocket.factory.js:159]: it can't https://docs.angularjs.org/api/ng/service/$interval

**Pull request merged**